File: compiler/bench/treebuild.fs
\ expected: 262143
\ Build tree recursively, count nodes (depth 18 = 262143 nodes)

: treebuild ( depth -- count ) recursive
  dup 0= if drop 1 exit then
  1- dup recurse swap recurse + 1+ ;

: main
  17 treebuild . cr ;

File: compiler/bench/treebuild.c
// expected: 262143
#include <stdio.h>
long treebuild(long depth) {
    if (depth == 0) return 1;
    return treebuild(depth - 1) + treebuild(depth - 1) + 1;
}

int main() {
    printf("%ld\n", treebuild(17));
    return 0;
}
